state-validate-util/src/main/java/org/sunbird/statevalidateutil/RecordProcessor.java

```diff
@@ -0,0 +1,206 @@
+     * @param requestParams
+     * @return
+     */
+    public static RecordProcessor getInstance(
+            ConnectionFactory connectionFactory, RequestParams requestParams) {
+        return new RecordProcessor(connectionFactory, requestParams);
+    }
+
+    /**
+     * this method is responsible to fetch the user external id data from cassandra and the table name
+     * is usr_external_identity and after fetching will convert the resultSet to List<User>
+     *
+     * @return List<User>
+     */
+    private List<User> getUserDataFromDbAsList() {
+        List<User> usersList = new ArrayList<>();
+        ResultSet resultSet = connection.getRecords("select * from user");
+        usersList.addAll(CassandraHelper.getUserListFromResultSet(resultSet));
+        return usersList;
+    }
+
+    /**
+     * this method will be used to process externalId
+     * this method will get the count of total number of records that need to be processed
+     *
+     * @return
+     */
+    public void startProcessingStateValidationAndFlags() throws IOException {
+        String custodianOrgId = getSystemSettingCustodianOrgId();
+        List<User> usersList = getRecordsToBeProcessed();
+        usersList
+                .stream()
+                .forEach(
+                        userObject -> {
+                            Map<String, String> compositeKeysMap = getCompositeKeysMap(userObject);
+                            try {
+                                startTracingRecord(userObject.getUserId());
+                                boolean stateValidated = isStateUSer(userObject.getUserId(), custodianOrgId);
+                                Map<String, Object> userBooleanMap = new HashMap<>();
+                                userBooleanMap.put(DbColumnConstants.stateValidated, stateValidated);
+                                userBooleanMap.put(DbColumnConstants.emailVerified, userObject.getEmailVerified());
+                                userBooleanMap.put(DbColumnConstants.phoneVerified, userObject.getPhoneVerified());
+                                userObject.setFlagsValue(calcFieldsValue(userBooleanMap));
+                                performSequentialOperationOnRecord(userObject, compositeKeysMap, stateValidated);
+                            } catch (Exception e) {
+                                logExceptionOnProcessingRecord(compositeKeysMap);
+                            } finally {
+                                endTracingRecord(userObject.getUserId());
+                            }
+                        });
+
+        connection.closeConnection();
+        closeWriterConnection();
+    }
+
+    private int calcFieldsValue(Map<String, Object> userBooleanMap) {
+        int userFlagValue = 0;
+        Set<Map.Entry<String, Object>> mapEntry = userBooleanMap.entrySet();
+        for(Map.Entry<String, Object> entry: mapEntry) {
+            if(StringUtils.isNotEmpty(entry.getKey()) && (Boolean) entry.getValue()) {
```

Comment:
i think in above scenario this can't be null.

Reply:
done.